projects/element-theme/src/styles/bootstrap/_button-group.scss

```diff
@@ -27,21 +38,49 @@
     border-end-end-radius: semantic-tokens.$element-button-radius;
   }
 
-  > .btn,
-  label > .btn {
-    position: relative;
-    flex: 1 1 auto;
-  }
-
   > .btn-check:focus + .btn,
-  > .btn:is(:hover, :active, .active) {
+  > .btn:is(:hover, :active, .active, :focus-visible) {
     z-index: 1;
   }
 
+  // Add separator border on primary button hover/focus to keep the layout consistent
+  > .btn-primary:has(+ .btn-primary):is(:hover, :active, .active, :focus-visible)::after {
+    content: '';
+    position: absolute;
+    inset-inline-end: 0;
+    border-inline-end: 1px solid var(--btn-color, inherit);
+    block-size: 100%;
+  }
+
   > .btn:not(:first-of-type),
   label + label {
     margin-inline-start: -1px;
   }
+
+  // Dropdown toggle styling for split buttons
+  .btn.dropdown-toggle {
+    inline-size: 24px;
+    padding: 0;
+    border-start-start-radius: 0;
+    border-end-start-radius: 0;
+
+    .dropdown-caret {
+      margin-inline: 0;
+      padding-inline: 0;
+      transform-origin: center;
+    }
+
+    &.show .dropdown-caret {
+      transform: rotate(180deg);
+    }
```

Comment:
This is not needed. `show` needs to be set on the dropdown-toggle, than it will rotate already automatically.